internalWebhook.go

```diff
@@ -43,21 +69,37 @@ func InternalWebhookToItem(now func() time.Time, iw InternalWebhook) (model.Item
 	}, nil
 }
 
-func ItemToInternalWebhook(i model.Item) (InternalWebhook, error) {
-	encodedWebhook, err := json.Marshal(i.Data)
-	if err != nil {
-		return InternalWebhook{}, err
+func ItemToInternalWebhook(i model.Item) (Register, error) {
+	var v1 RegistryV1
+	var v2 RegistryV2
+	encodedWebhook, e := json.Marshal(i.Data)
+	if e != nil {
+		return nil, e
 	}
-	var iw InternalWebhook
-	err = json.Unmarshal(encodedWebhook, &iw)
-	if err != nil {
-		return InternalWebhook{}, err
+
+	for v := range i.Data {
```

Comment:
I recommend going with something like this instead of the for loop

```go
			e = json.Unmarshal(encodedWebhook, &v1)
			if e == nil {
				return v1, nil
			}
			
			e = json.Unmarshal(encodedWebhook, &v2)
			if e == nil {
				return v2, nil
			}
```